tests/functional/object/mcg/test_sts_client.py

```diff
@@ -0,0 +1,258 @@
+
+    request.addfinalizer(teardown)
+    return factory
+
+
+@pytest.fixture()
+def nb_account_factory(request):
+    """
+    Fixture to create noobaa account
+
+    """
+    nb_account = []
+
+    def factory(mcg, user_name, user_email, **kwargs):
+        """
+        Factory function to create the noobaa account
+
+        Args:
+            mcg (MCG): MCG object
+            user_name (str): User name
+            user_email (str): Email of the user
+
+        """
+        account = NoobaaAccount(
+            mcg=mcg, name=f"{user_name}", email=f"{user_email}", **kwargs
+        )
+        nb_account.append(account)
+
+        return account
+
+    def teardown():
+        """
+        Teardown function that deletes the Noobaa account
+
+        """
+        for account in nb_account:
+            account.delete_account()
+
+    request.addfinalizer(teardown)
+    return factory
+
+
+@pytest.fixture()
+def new_bucket(request, mcg_obj_session):
+    """
+    Create new bucket using s3
+
+    """
+    buckets_created = []
+
+    def factory(bucket_name, s3_client=None):
+        s3_create_bucket(mcg_obj_session, bucket_name, s3_client)
+        logger.info(f"Created new-bucket {bucket_name}")
+        bucket_obj = {
+            "s3client": s3_client,
+            "bucket": bucket_name,
+            "mcg": mcg_obj_session,
+        }
+        buckets_created.append(bucket_obj)
+
+    def finalizer():
+        """
+        Cleanup the created bucket
+
+        """
+        for bucket in buckets_created:
+            logger.info(f"Deleting bucket {bucket.get('bucket')}")
+            s3_delete_bucket(
+                bucket.get("mcg"), bucket.get("bucket"), bucket.get("s3client")
+            )
+            logger.info(f"Deleted the bucket {bucket}")
+
+    request.addfinalizer(finalizer)
+    return factory
+
+
+@mcg
+@red_squad
+@tier2
+class TestSTSClient:
+    def test_sts_assume_role(
+        self,
+        mcg_obj_session,
+        awscli_pod_session,
+        nb_account_factory,
+        nb_assign_user_role_fixture,
+        new_bucket,
+        add_env_vars_to_noobaa_core_class,
+        add_env_vars_to_noobaa_endpoint_class,
+    ):
+        """
+        Test sts support for Noobaa clients.
+        As part of this, we test the following:
+            * Assign role to an user
+            * Assume the role of an user
+            * Perform IO by assuming of the role of another user
+            * Remove the role from the user
+            * Try to assume the role
+
+        """
+        # change sts token expire time to 10 minute
+        add_env_vars_to_noobaa_core_class(
+            [(constants.STS_DEFAULT_SESSION_TOKEN_EXPIRY_MS, 600000)]
+        )
+        logger.info("Changed the sts token expiration time to 10 minutes")
+
+        add_env_vars_to_noobaa_endpoint_class(
+            [(constants.STS_DEFAULT_SESSION_TOKEN_EXPIRY_MS, 600000)]
+        )
+        logger.info("Changed the sts token expiration time to 10 minutes")
```

Comment:
Why do we need this step? At the final step of the test where we're waiting for the role-assumption to no longer work, it looks like we're expecting that because we removed the role, and not because the token expired

Reply:
@sagihirshfeld once you remove the role config from a user.. it no longer allows new role assumption. But the old token would still be valid until the expiration is reached

Reply:
Ohhh ok thanks for the clarification 👍